app-backend/src/main/kotlin/org/example/appbackend/entity/User.kt

```diff
@@ -0,0 +1,44 @@
+package org.example.appbackend.entity
+
+import jakarta.persistence.*
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Entity
+@Table(name = "users")
+class User {
+        @Id @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "user_sequence")
+        @SequenceGenerator(name = "user_sequence", sequenceName = "user_sequence", allocationSize = 1)
+        @Column(name = "id")
+        val id: Int? = null
+
+        @Column(name = "username")
+        var username: String = ""
```

Comment:
Are you sure that spring is able to infer db nullability from the type? Should not be there additional field in annotation?